src/fmeval/eval_algorithms/faithfulness.py

```diff
@@ -0,0 +1,308 @@
+            input_keys=[input_key],
+            output_keys=[output_key],
+        )
+        self.input_key = input_key
+        self.output_key = output_key
+        self.judge_model = judge_model
+
+    @validate_call
+    def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
+        """Invokes the judge model to generate statements and augments the input record with the statements.
+        :param record: The input record.
+        :returns: The input record with the statements added in.
+        """
+        get_raw_statements = GetModelOutputs(
+            input_to_output_keys={self.input_key: [RAW_STATEMENTS]},
+            model_runner=self.judge_model,
+        )
+        record_with_raw_statements = get_raw_statements(record)
+        raw_statements = record_with_raw_statements[RAW_STATEMENTS].split("\n")
+        list_statements = [statement for statement in raw_statements if statement.startswith("Statement:")]
+        statements_str: str = "\n".join([f"{i + 1}. {st}" for i, st in enumerate(list_statements)])
+        record[self.output_key] = statements_str
+        return record
+
+
+class Faithfulness(EvalAlgorithmInterface):
+    """
+    This evaluation measures the factual consistency of the generated answer against the given context.
+    It is calculated from answer and retrieved context. The answer is scaled to (0,1) range. Higher the better.
+
+    The generated answer is regarded as faithful if all the claims that are made in the answer can be inferred from
+    the given context. To calculate this a set of claims from the generated answer is first identified.
+    Then each one of these claims are cross checked with given context to determine if it can be inferred from given
+    context or not.
+    """
+
+    eval_name = EvalAlgorithm.FAITHFULNESS.value
+
+    def __init__(self):
+        """Faithfulness initializer."""
+        super().__init__(EvalAlgorithmConfig())
+
+    @staticmethod
+    def _build_pipeline(
+        judge_model: ModelRunner,
+        long_form_prompt_template: str,
+        nli_statements_prompt_template: str,
+    ) -> TransformPipeline:
+        gen_long_form_prompt = GeneratePrompt(
+            input_keys=[],
+            output_keys=[LONG_FORM_PROMPT],
+            prompt_template=long_form_prompt_template,
+            placeholder_to_record_key={
+                QUESTION: DatasetColumns.MODEL_INPUT.value.name,
+                ANSWER: DatasetColumns.MODEL_OUTPUT.value.name,
+            },
+        )
+        get_statements = GetStatements(
+            input_key=LONG_FORM_PROMPT,
+            output_key=STATEMENTS,
+            judge_model=judge_model,
+        )
+        gen_nli_statements_prompt = GeneratePrompt(
+            input_keys=[],
+            output_keys=[NLI_STATEMENTS_PROMT],
+            prompt_template=nli_statements_prompt_template,
+            placeholder_to_record_key={"context": DatasetColumns.TARGET_CONTEXT.value.name, STATEMENTS: STATEMENTS},
```

Comment:
Note for other reviewers: the `STATEMENTS: STATEMENTS` is not a typo. We have `$statements` in the NLI statements prompt template (this is what the first `STATEMENTS` corresponds to), while the second `STATEMENTS` corresponds to the key that stores the statements generated by the `get_statements` transform above. @xiaoyi-cheng maybe worth adding a comment for this, but this is non-blocking.